File: 121.买卖股票的最佳时机.go
/*
 * @lc app=leetcode.cn id=121 lang=golang
 *
 * [121] 买卖股票的最佳时机
 */

// @lc code=start
func maxProfit(prices []int) int {
	max := 0
	min := prices[0]
	for i := 0; i < len(prices); i++ {
		if prices[i] < min {
			min = prices[i]
		} else if dif := prices[i] - min; dif > max {
			max = prices[i] - min
		}
	}
	return max
}

// @lc code=end

File: 121.买卖股票的最佳时机.py
#
# @lc app=leetcode.cn id=121 lang=python3
#
# [121] 买卖股票的最佳时机
#

# @lc code=start
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max = 0 
        min = prices[0]
        for _, val in enumerate(prices):
            if val < min:
                min = val
            else:
                if val - min > max:
                    max = val - min
        return max
# @lc code=end

